projects/plugins/jetpack/extensions/store/wordpress-com/reducer.ts

```diff
@@ -0,0 +1,93 @@
+/**
+ * Types
+ */
+import {
+	ACTION_INCREASE_AI_ASSISTANT_REQUESTS_COUNT,
+	ACTION_REQUEST_AI_ASSISTANT_FEATURE,
+	ACTION_SET_PLANS,
+	ACTION_STORE_AI_ASSISTANT_FEATURE,
+} from './constants';
+import type { PlanStateProps } from './types';
+
+const INITIAL_STATE: PlanStateProps = {
+	plans: [],
+	features: {
+		aiAssistant: {
+			hasFeature: false,
+			isOverLimit: false,
+			requestsCount: 0,
+			requestsLimit: 0,
+			requireUpgrade: false,
+			errorMessage: '',
+			errorCode: '',
+			upgradeType: 'default',
+			currentTier: {
+				value: 1,
+			},
+			usagePeriod: {
+				currentStart: '',
+				nextStart: '',
+				requestsCount: 0,
+			},
+			_meta: {
+				isRequesting: false,
+			},
+		},
+	},
+};
+
+export default function reducer( state = INITIAL_STATE, action ) {
+	switch ( action.type ) {
+		case ACTION_SET_PLANS:
+			return {
+				...state,
+				plans: action.plans,
+			};
+
+		case ACTION_REQUEST_AI_ASSISTANT_FEATURE:
+			return {
+				...state,
+				features: {
+					...state.features,
+					aiAssistant: {
+						...state.features.aiAssistant,
+						_meta: {
+							...state.features.aiAssistant._meta,
+							isRequesting: true,
+						},
+					},
+				},
+			};
+
+		case ACTION_STORE_AI_ASSISTANT_FEATURE: {
+			return {
+				...state,
+				features: {
+					...state.features,
+					aiAssistant: {
+						...action.feature,
+						_meta: {
+							...state.features.aiAssistant._meta,
+							isRequesting: false,
+						},
+					},
+				},
+			};
+		}
+
+		case ACTION_INCREASE_AI_ASSISTANT_REQUESTS_COUNT: {
+			return {
```

Comment:
Shouldn't this also increase the `features.aiAssistant.usagePeriod.requestsCount` number? Or maybe we have some useless redundancy there?

I guess `aiAssistant.requestsCount` was the original one, used for free and all plans, and now we seem to be moving into this specific usage object. For now, I'd say we can update both in the same reducer?

Reply:
Either that or decide which one is it and get rid of the other, but I feel this should be done on a separate PR

Reply:
I think we should increase both. Nice feedback

Reply:
Did we ended up not increment the `features.aiAssistant.usagePeriod.requestsCount` number? I think it will be important on usage meter for the tiers.